packager/amp_cache_transform/amp_cache_transform.go

```diff
@@ -71,13 +80,102 @@ func parseParameterisedIdentifier(reader *strings.Reader) (*parameterisedIdentif
 	if err != nil {
 		return nil, errors.Wrap(err, "parsing primary identifier")
 	}
-	// NOTE: The initial version of AMP-Cache-Transform does not support
-	// any parameters, so we don't bother to implement a parser for them.
-	// No point in distinguishing syntactic failure from semantic failure.
-	// Once we add parameters, we'll need to implement parsers for the
-	// subset item that they use. See Items here:
-	// https://tools.ietf.org/html/draft-ietf-httpbis-header-structure-07#section-3.4
-	return &parameterisedIdentifier{primaryID}, nil
+	// NOTE: The current version of AMP-Cache-Transform only uses
+	// string-valued parameters. Future versions of the spec may require
+	// parsing other Item types.
+	params := map[string]string{}
+	for {
+		if err := discardOWS(reader); err != nil {
+			return nil, errors.Wrap(err, "discarding OWS")
+		}
+
+		// If we've reach the end of the parameter list, exit success.
+		if reader.Len() == 0 {
+			break
+		}
+		semicolon, err := reader.ReadByte()
```

Comment:
Might be more readable to use the if initializer, e.g. 
if semicolor, err := reader.ReadByte(); err == nil {
    if semicolon != ';' {
    }
} else {
    return nil, errors.Wrap(....)
}

This keeps variable scope as minimal as possible - I don't see semicolon being referenced later.

Reply:
Done.